content/blog/hpe-oneview-and-servicenow-integration-with-stackstorm.md

```diff
@@ -0,0 +1,46 @@
+
+This would involve writing the code for **both systems**. What if we leveraged Stackstorm? A quick check of the Stackstorm exchange [exchange.stackstorm.com](exchange.stackstorm.com) and we find that there is Stackstorm integration pack available for ServiceNow. If we use Stackstorm, we only have to write **half** the code. All that needs done is to write the code for a HPE OneView Stackstorm integration pack. 
+
+I have written a couple of other blogs on Stackstorm and if you are interested in trying this approach, I suggest you go to <https://developer.hpe.com/blog/tag/stackstorm> and read my other posts.
+
+Developing the **stackstorm-hpe-oneview** integration pack, which is available here: <https://github.com/HewlettPackard/stackstorm-hpe-oneview>, is fairly straightforward. For this integration to function we will have to write five very short actions and a couple of simple rules. You can see in the chart at the top that two of the actions will be used with the first workflow and three will be need for the second workflow. Actions are the workhorse of Stackstorm. They are basically the same thing you would write to accomplish this task. All that needs done is to  add just a few lines of code to turn them into Stackstorm actions. Let's have a look!
+
+![](http://www.techworldwookie.com/blogpost/action.png)
+
+The flow of the above code is simple. This is a script that gets the alarms from HPE OneView, checks to see if it is a list and if so, returns it. Super simple. The second action in workflow "A" will format the information into a mongo record, add a process field and save the mongo document. Again, this is very simple to code and test. The class is passed the alarms and iterates through each one, a query to check if the document exists and if not, formats a dictionary and writes the document via pymongo. This is all it takes to collect the alarms and save them in the database. 
+
+![](http://www.techworldwookie.com/blogpost/load.png "Load database example")
+
+The second workflow, workflow "B" will call another action every five minutes that reads the documents from the mongo database, looks for the processed flag set to no, collects the results into a python list and returns it. 
+
+![](http://www.techworldwookie.com/blogpost/get-records.png "Get documents from mongo")
+
+The next task is to send the list of alarms to ServiceNow. Here is where the power of integration packs comes into view. All we need to do is issue a command on the Stackstorm server **"st2 pack install servicenow"**. By issuing this command we gain access to the automation scripts (actions) that are pre-written. This makes the job much easier. What if I wanted to integrate Twitter into my automation flow? Easy, **st2 pack install twitter**.  I won't show you the ServiceNow action script here but you can look at if on the exchange if you like here: <https://github.com/StackStorm-Exchange/stackstorm-servicenow/blob/master/actions/create_record.py>
```

Comment:
"Stackstorm" should be "StackStorm"
Suggest rewrite from the royal "we" for these two sentences:
All you need to do is issue...
By issuing this command, you gain access...

Highly recommend breaking this paragraph out into two paragraphs, starting the second paragraph with:
"What if I wanted to integrate Twitter into my automation flow?

End "if you like here" with a period and hotlink "here" instead of spelling out the URL